src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,82 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+interface TodoModalProps {
+  todo: Todo;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({ todo }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
```

Comment:
Ensure that the `getUser` function is correctly implemented in the `../../api` module. This function should return a promise that resolves with the user data based on the `userId`.